Source/Project64-audio/Driver/DirectSound.cpp

```diff
@@ -132,10 +132,16 @@ void DirectSoundDriver::SetFrequency(uint32_t Frequency)
 
 void DirectSoundDriver::SetVolume(uint32_t Volume)
 {
-    /*DWORD dsVolume = ((DWORD)Volume * -25);
-    if (Volume == 100) dsVolume = (DWORD)DSBVOLUME_MIN;
-    if (Volume == 0) dsVolume = DSBVOLUME_MAX;
-    if (lpdsb != NULL) lpdsb->SetVolume(dsVolume);*/
+    LPDIRECTSOUNDBUFFER & lpdsb = (LPDIRECTSOUNDBUFFER &)m_lpdsb;
+    int32_t dsVolume = - 10 ^ ((100 - Volume) * 33);
```

Comment:
I recognize that you intend to use the logarithmic solution from that article you linked to.

But you do realize that `-10 ^ ((100 - volume) * 33)` is not the same as `pow(-10, (100 - volume) * 33)`?

I seriously doubt you intended to use the bit-wise XOR operation in C to simulate a logarithmic scale effect, because that's what the `^` operator in C does.  XOR would just completely scramble the result of 33*(100 - v) by the binary mask 1010 and has no mathematical significance towards the result's exponentiation.